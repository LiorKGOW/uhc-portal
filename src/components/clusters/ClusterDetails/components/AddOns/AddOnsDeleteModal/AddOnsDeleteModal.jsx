import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Form, TextInput } from '@patternfly/react-core';

import Modal from '../../../../../common/Modal/Modal';

import ErroBox from '../../../../../common/ErrorBox';

import '../AddOns.scss';

class AddOnsDeleteModal extends Component {
  state = {
    addOnNameInput: '',
  }

  componentDidUpdate(prevProps) {
    const { deleteClusterAddOnResponse } = this.props;
    if (!prevProps.deleteClusterAddOnResponse.fulfilled && deleteClusterAddOnResponse.fulfilled) {
      this.handleClose();
    }
  }

  setValue = (newInput) => {
    this.setState({
      addOnNameInput: newInput,
    });
  }

  handleClose = () => {
    const { closeModal } = this.props;
    this.setState({
      addOnNameInput: '',
    });
    closeModal();
  }

  render() {
    const {
      isOpen,
      modalData,
      deleteClusterAddOn,
      deleteClusterAddOnResponse,
    } = this.props;

    const {
      addOnNameInput,
    } = this.state;

    const {
      addOnName,
      addOnID,
      clusterID,
    } = modalData;

    const isValid = addOnNameInput === addOnName;

    const handleSubmit = () => {
      deleteClusterAddOn(clusterID, addOnID);
    };

    const submitForm = (e) => {
      e.preventDefault();
      if (isValid) {
        handleSubmit();
      }
    };

    const errorContainer = deleteClusterAddOnResponse.error && (
      <ErroBox message="Error uninstalling add-on" response={deleteClusterAddOnResponse} />
    );

    const isPending = deleteClusterAddOnResponse.pending;

    return isOpen && (
    <Modal
      title={`Uninstall ${addOnName}`}
      onClose={this.handleClose}
      primaryText="Uninstall"
      primaryVariant="danger"
      isPrimaryDisabled={!isValid}
      onPrimaryClick={handleSubmit}
      onSecondaryClick={this.handleClose}
      isPending={isPending}
    >
      <p>
        {errorContainer}
            This action will uninstall the add-on,
            removing add-on data from cluster can not be undone.
      </p>
      <Form onSubmit={submitForm}>
        <p>
              Confirm deletion by typing
          {' '}
          <span className="addon-delete-modal-textinput">{addOnName}</span>
          {' '}
              below:
        </p>
        <TextInput
          type="text"
          value={addOnNameInput}
          placeholder="Enter name"
          onChange={this.setValue}
          aria-label="addon name"
        />
      </Form>
    </Modal>
    );
  }
}

AddOnsDeleteModal.propTypes = {
  isOpen: PropTypes.bool.isRequired,
  modalData: PropTypes.shape({
    addOnName: PropTypes.string,
    addOnID: PropTypes.string,
    clusterID: PropTypes.string,
  }),
  closeModal: PropTypes.func.isRequired,
  deleteClusterAddOn: PropTypes.func.isRequired,
  deleteClusterAddOnResponse: PropTypes.object.isRequired,
};

export default AddOnsDeleteModal;
