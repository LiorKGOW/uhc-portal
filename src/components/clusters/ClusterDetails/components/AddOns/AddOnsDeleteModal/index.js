import { connect } from 'react-redux';
import shouldShowModal from '../../../../../common/Modal/ModalSelectors';
import { closeModal } from '../../../../../common/Modal/ModalActions';
import AddOnsDeleteModal from './AddOnsDeleteModal';
import { addOnsActions } from '../AddOnsActions';


const mapStateToProps = state => ({
  isOpen: shouldShowModal(state, 'add-ons-delete-modal'),
  modalData: state.modal.data,
  deleteClusterAddOnResponse: state.addOns.deleteClusterAddOnResponse,
});

const mapDispatchToProps = {
  deleteClusterAddOn: (clusterID, addOnID) => addOnsActions.deleteClusterAddOn(clusterID, addOnID),
  closeModal: () => closeModal(),
  clearDeleteAddOnResponse: () => addOnsActions.clearClusterAddOnsResponses(),
};

export default connect(mapStateToProps, mapDispatchToProps)(AddOnsDeleteModal);
