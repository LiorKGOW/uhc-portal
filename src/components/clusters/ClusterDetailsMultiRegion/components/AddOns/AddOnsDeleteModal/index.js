import { connect } from 'react-redux';

import { closeModal } from '../../../../../common/Modal/ModalActions';
import shouldShowModal from '../../../../../common/Modal/ModalSelectors';
import { addOnsActions, clearClusterAddOnsResponses } from '../AddOnsActions';

import AddOnsDeleteModal from './AddOnsDeleteModal';

const mapStateToProps = (state) => ({
  isOpen: shouldShowModal(state, 'add-ons-delete-modal'),
  modalData: state.modal.data,
  deleteClusterAddOnResponse: state.addOns.deleteClusterAddOnResponse,
});

const mapDispatchToProps = (dispatch) => ({
  deleteClusterAddOn: (clusterID, addOnID) =>
    dispatch(addOnsActions.deleteClusterAddOn(clusterID, addOnID)),
  closeModal: () => dispatch(closeModal()),
  clearClusterAddOnsResponses: () => dispatch(clearClusterAddOnsResponses()),
});

export default connect(mapStateToProps, mapDispatchToProps)(AddOnsDeleteModal);
