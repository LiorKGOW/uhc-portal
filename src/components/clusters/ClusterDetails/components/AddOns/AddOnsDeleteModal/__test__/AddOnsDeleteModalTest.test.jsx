import React from 'react';
import { shallow } from 'enzyme';
import { TextInput } from '@patternfly/react-core';

import AddOnsDeleteModal from '../AddOnsDeleteModal';

describe('<AddOnsDeleteModal />', () => {
  let wrapper;
  let closeModal;
  let deleteClusterAddOn;
  let clearClusterAddOnsResponses;
  beforeAll(() => {
    closeModal = jest.fn();
    deleteClusterAddOn = jest.fn();
    clearClusterAddOnsResponses = jest.fn();
    wrapper = shallow(
      <AddOnsDeleteModal
        isOpen
        modalData={{
          addOnName: 'fake-addon-name',
          addOnID: 'fake-addon-id',
          clusterID: 'fake-cluster-id',
        }}
        closeModal={closeModal}
        deleteClusterAddOn={deleteClusterAddOn}
        clearClusterAddOnsResponses={clearClusterAddOnsResponses}
        deleteClusterAddOnResponse={{ fulfilled: false, pending: false, error: false }}
      />,
    );
  });
  it('renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('delete button should be enabled only after inputting the add on name', () => {
    expect(wrapper.find('Modal').props().isPrimaryDisabled).toBeTruthy(); // disabled at first

    wrapper.find(TextInput).simulate('change', 'fake');
    expect(wrapper.find('Modal').props().isPrimaryDisabled).toBeTruthy(); // disabled when unrelated input

    wrapper.find(TextInput).simulate('change', 'fake-addon-name');
    expect(wrapper.find('Modal').props().isPrimaryDisabled).toBeFalsy(); // enabled when correct
  });

  it('should close modal on cancel', () => {
    const modal = wrapper.find('Modal');
    modal.props().onSecondaryClick();
    expect(closeModal).toBeCalled();
  });

  it('should call deleteClusterAddOn correctly', () => {
    const modal = wrapper.find('Modal');
    modal.props().onPrimaryClick();
    expect(deleteClusterAddOn).toBeCalledWith('fake-cluster-id', 'fake-addon-id');
  });

  it('should close correctly on succeess', () => {
    wrapper.setProps({
      deleteClusterAddOnResponse: { fulfilled: true, pending: false, error: false },
    });
    expect(closeModal).toBeCalled();
  });

  it('should clear response when modal is closed', () => {
    const modal = wrapper.find('Modal');
    modal.props().onSecondaryClick();
    expect(clearClusterAddOnsResponses).toBeCalled();
  });
});
