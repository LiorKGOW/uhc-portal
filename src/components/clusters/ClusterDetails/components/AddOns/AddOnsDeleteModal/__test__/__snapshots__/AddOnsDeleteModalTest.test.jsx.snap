// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AddOnsDeleteModal /> renders correctly 1`] = `
<Modal
  isPending={false}
  isPrimaryDisabled={true}
  onClose={[Function]}
  onPrimaryClick={[Function]}
  onSecondaryClick={[Function]}
  primaryText="Uninstall"
  primaryVariant="danger"
  title="Uninstall fake-addon-name"
>
  <p>
    This action will uninstall the add-on, removing add-on data from cluster can not be undone.
  </p>
  <Form
    onSubmit={[Function]}
  >
    <p>
      Confirm deletion by typing
       
      <span
        className="addon-delete-modal-textinput"
      >
        fake-addon-name
      </span>
       below:
    </p>
    <TextInput
      aria-label="addon name"
      onChange={[Function]}
      placeholder="Enter name"
      type="text"
      value=""
    />
  </Form>
</Modal>
`;
